packages/storage/src/reference.ts

```diff
@@ -132,224 +107,248 @@ export class Reference {
     return this.service;
   }
 
-  /**
-   * Uploads a blob to this object's location.
-   * @param data The blob to upload.
-   * @return An UploadTask that lets you control and
-   *     observe the upload.
-   */
-  put(
-    data: Blob | Uint8Array | ArrayBuffer,
-    metadata: Metadata | null = null
-  ): UploadTask {
-    validate('put', [uploadDataSpec(), metadataSpec(true)], arguments);
-    this.throwIfRoot_('put');
-    return new UploadTask(
-      this,
-      this.service,
-      this.location,
-      this.mappings(),
-      new FbsBlob(data),
-      metadata
-    );
-  }
-
-  /**
-   * Uploads a string to this object's location.
-   * @param value The string to upload.
-   * @param format The format of the string to upload.
-   * @return An UploadTask that lets you control and
-   *     observe the upload.
-   */
-  putString(
-    value: string,
-    format: StringFormat = StringFormat.RAW,
-    metadata?: Metadata
-  ): UploadTask {
-    validate(
-      'putString',
-      [stringSpec(), stringSpec(formatValidator, true), metadataSpec(true)],
-      arguments
-    );
-    this.throwIfRoot_('putString');
-    const data = dataFromString(format, value);
-    const metadataClone = Object.assign({}, metadata);
-    if (
-      !type.isDef(metadataClone['contentType']) &&
-      type.isDef(data.contentType)
-    ) {
-      metadataClone['contentType'] = data.contentType!;
+  throwIfRoot_(name: string): void {
+    if (this.location.path === '') {
+      throw errorsExports.invalidRootOperation(name);
     }
-    return new UploadTask(
-      this,
-      this.service,
-      this.location,
-      this.mappings(),
-      new FbsBlob(data.data, true),
-      metadataClone
-    );
   }
+}
 
-  /**
-   * Deletes the object at this location.
-   * @return A promise that resolves if the deletion succeeds.
-   */
-  delete(): Promise<void> {
-    validate('delete', [], arguments);
-    this.throwIfRoot_('delete');
-    return this.service.getAuthToken().then(authToken => {
-      const requestInfo = requests.deleteObject(this.service, this.location);
-      return this.service.makeRequest(requestInfo, authToken).getPromise();
-    });
-  }
+/**
+ * Uploads a blob to this object's location.
+ * @param data The blob to upload.
+ * @return An UploadTask that lets you control and
+ *     observe the upload.
+ */
+export function uploadBytes(
+  ref: Reference,
+  data: Blob | Uint8Array | ArrayBuffer,
+  metadata: Metadata | null = null
+): UploadTask {
+  validate('put', [uploadDataSpec(), metadataSpec(true)], arguments);
```

Comment:
Should we simplify the input validation to not check number of arguments and the type of arguments?

Reply:
Note that if we do drop it for the new SDK, we should probably add it to the compatibility layer.